zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/UserTaskTest.java

```diff
@@ -0,0 +1,171 @@
+  protected ZeebeClient client;
+
+  @BeforeAll
+  static void init() {
+    CLUSTER.start().awaitCompleteTopology();
+  }
+
+  @BeforeEach
+  void initClientAndInstances() {
+    final var gateway = CLUSTER.anyGateway();
+    client =
+        CLUSTER
+            .newClientBuilder()
+            .gatewayAddress(gateway.gatewayAddress())
+            .gatewayRestApiPort(gateway.mappedPort(TestZeebePort.REST))
+            .defaultRequestTimeout(Duration.ofSeconds(15))
+            .build();
+    userTaskKey = createSingleUserTask();
+  }
+
+  protected long createSingleUserTask() {
+    return createUserTasks(b -> {}, "{}", 1).getFirst();
+  }
+
+  protected List<Long> createUserTasks(
+      final Consumer<UserTaskBuilder> consumer, final String variables, final int amount) {
+
+    final BpmnModelInstance modelInstance = createSingleUserTaskModelInstance(consumer);
+    final long processDefinitionKey = deployProcess(modelInstance);
+
+    final var processInstanceKeys =
+        IntStream.range(0, amount)
+            .boxed()
+            .map(i -> createProcessInstance(processDefinitionKey, variables))
+            .toList();
+
+    final List<Long> userTaskKeys =
+        RecordingExporter.userTaskRecords(UserTaskIntent.CREATED)
+            .filter(r -> processInstanceKeys.contains(r.getValue().getProcessInstanceKey()))
+            .limit(amount)
+            .map(Record::getKey)
+            .collect(Collectors.toList());
+
+    assertThat(userTaskKeys).describedAs("Expected %d created user tasks", amount).hasSize(amount);
+
+    return userTaskKeys;
+  }
+
+  protected long deployProcess(final BpmnModelInstance modelInstance) {
+    final DeploymentEvent deploymentEvent =
+        client
+            .newDeployResourceCommand()
+            .addProcessModel(modelInstance, "process.bpmn")
+            .send()
+            .join();
+    //    waitUntilDeploymentIsDone(deploymentEvent.getKey());
+    return deploymentEvent.getProcesses().get(0).getProcessDefinitionKey();
+  }
+
+  protected BpmnModelInstance createSingleUserTaskModelInstance(
+      final Consumer<UserTaskBuilder> taskBuilderConsumer) {
+    return Bpmn.createExecutableProcess("process")
+        .startEvent("start")
+        .userTask("task", taskBuilderConsumer)
+        .zeebeUserTask()
+        .endEvent("end")
+        .done();
+  }
+
+  protected long createProcessInstance(final long processDefinitionKey, final String variables) {
+    return client
+        .newCreateInstanceCommand()
+        .processDefinitionKey(processDefinitionKey)
+        .variables(variables)
+        .send()
+        .join()
+        .getProcessInstanceKey();
+  }
+
+  protected long createProcessInstance(final long processDefinitionKey) {
+    return client
+        .newCreateInstanceCommand()
+        .processDefinitionKey(processDefinitionKey)
+        .send()
+        .join()
+        .getProcessInstanceKey();
+  }
+
+  protected void waitUntilDeploymentIsDone(final long key) {
```

Comment:
💭 This doesn't seem to be used anywhere, or am I missing something? The only usage is commented out on the line 110

Reply:
Good catch, yeah 👍 I'll adjust the base test class a bit and remove unused stuff 🙂

Reply:
Done with https://github.com/camunda/zeebe/pull/16683/commits/cfeefeba058923a0260ceaae5542af031c6a5df4